__tests__/login/oidc/ClientRegistrar.spec.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import "reflect-metadata";
+import { FetcherMock } from "../../../src/util/__mocks__/Fetcher";
+import ClientRegistrar from "../../../src/login/oidc/ClientRegistrar";
+import { IssuerConfigFetcherFetchConfigResponse } from "../../../src/login/oidc/__mocks__/IssuerConfigFetcher";
+import { Response as NodeResponse } from "node-fetch";
```

Comment:
Why use the `node-fetch` `Response` type here, instead of `lib.dom.d.ts` `Response` ? A mix of these two types appear in other places in the code, and I'm curious about the motivation for it.

Reply:
If I understand it correctly, in this case it's because it's not a type but a class, i.e. `new NodeResponse` is specifically executed below, and unfortunately, it does not appear to be compatible with the browser's Response API?

Reply:
Exactly. You can't instantiate Response in lib.dom.d.ts.

Reply:
Thanks !